### PR TITLE
Updated configuration class for check output_type for different adapters. Cosmetic changes

### PR DESCRIPTION
**Pull request by fanantoxa**
@soulcutter I've updated configuration class. Should fix:
> disallow Saxerator::Configuration#output_type = :xml for Ox (or alternately get that to work, but that seems way harder at this point)

Take a look please when you'll find a time.

### CODE REVIEW

**Comment by fanantoxa** [comment]
@soulcutter Also updated benchmark file for test with all adapters

**Comment by fanantoxa** [comment]
@soulcutter Did you saw this requests ? :)

**Comment by soulcutter** [comment]
Yes, I am going to a hack night tomorrow and will be looking at this. Apologies for the silence, have been working very hard lately

Bradley Schaefer

**Comment by fanantoxa** [comment]
@soulcutter It's ok) Waiting for your feedback)

**Comment by fanantoxa** [comment]
@soulcutter Maybe we can find another way for communication? Maybe skype / slack or gitter to avoid this ping questions and faster communication?

**Comment by soulcutter** [comment]
I'm Sou|cutter on FreeNode IRC

File: lib/saxerator/configuration.rb [hunk 1]
```diff
@@ -69,8 +78,14 @@ def put_attributes_in_hash?
...
     def raise_error_if_using_put_attributes_in_hash_with_xml
       if @output_type != :hash && @put_attributes_in_hash
-        raise ArgumentError.new("put_attributes_in_hash! is only valid when using output_type = :hash (the default)'")
+        raise_exeption("put_attributes_in_hash! is only valid when using output_type = :hash (the default)'")
       end
     end
+
+    private
+
+    def raise_exeption(text)
```
Comment (soulcutter) [inline lib/saxerator/configuration.rb]: This is misspelled `raise_exception` but also, I don't think this extraction is necessary. `raise ArgumentError, ""` is not much duplication (or we'd have these sorts of methods all over the place)

File: benchmark/benchmark.rb [hunk 2]
```diff
@@ -9,15 +9,58 @@
...
+  end
 end
 
-puts "for_tag:   #{count} artist elements parsed"
-puts "at_depth:  #{count2} elements parsed"
-puts "within:    #{count3} artists children parsed"
-puts "composite: #{count4} names within artist nested 3 tags deep parsed"
+saxerator_benchmark = SaxeratorBenchmark.new(file)
+
+ADAPTERS.each {|adapter| saxerator_benchmark.with_adapter(adapter) }
```
Comment (soulcutter) [inline benchmark/benchmark.rb]: inconsistent whitespace here.

File: benchmark/benchmark.rb [hunk 3]
```diff
@@ -9,15 +9,58 @@
...
+    puts "composite: #{count4} names within artist nested 3 tags deep parsed"
+    puts
+  end
 end
 
-puts "for_tag:   #{count} artist elements parsed"
-puts "at_depth:  #{count2} elements parsed"
-puts "within:    #{count3} artists children parsed"
-puts "composite: #{count4} names within artist nested 3 tags deep parsed"
+saxerator_benchmark = SaxeratorBenchmark.new(file)
```
Comment (soulcutter) [inline benchmark/benchmark.rb]: This is awesome that you updated the benchmarks!

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Benchmark with :nokogiri parser

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

       user     system      total        real
for_tag  5.800000   0.020000   5.820000 (  5.823733)
at_depth  4.590000   0.020000   4.610000 (  4.605400)
within  5.640000   0.010000   5.650000 (  5.657823)
composite  4.310000   0.020000   4.330000 (  4.328335)

##########################################################

for_tag:   100000 artist elements parsed
at_depth:  400000 elements parsed
within:    100000 artists children parsed
composite: 0 names within artist nested 3 tags deep parsed

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Benchmark with :ox parser

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

       user     system      total        real
for_tag  3.710000   0.010000   3.720000 (  3.723125)
at_depth  2.710000   0.010000   2.720000 (  2.723924)
within  3.630000   0.010000   3.640000 (  3.644329)
composite  2.130000   0.010000   2.140000 (  2.139019)

##########################################################

for_tag:   100000 artist elements parsed
at_depth:  400000 elements parsed
within:    100000 artists children parsed
composite: 0 names within artist nested 3 tags deep parsed
```

Seems like the composite one is slightly broken, but that's unrelated to this changeset. Ox is faster 👍

**Comment by soulcutter** [comment]
I like this, but it will need rebasing off of my ox branch and tweaked a bit for adapter specific tests based on my spec helper changes.

Thanks a ton for your contributions!

**Comment by fanantoxa** [comment]
@soulcutter I've added you on IRC, please let me know when you'll available I want to discuss this and few other ideas.

**Comment by fanantoxa** [comment]
@soulcutter I've rebased form master and updated code with your additions. Check please

**Comment by fanantoxa** [comment]
@soulcutter sorry can't find you on freenode, say that No such nick/channel. could you find me with nick 'fanantoxa' please ?

**Comment by fanantoxa** [comment]
@soulcutter fixed all additions that you've mentioned. All tests pass

**Comment by soulcutter** [comment]
I think we need to open this against the `master` branch

**Comment by fanantoxa** [comment]
@soulcutter  I'll rebaise from `master` again and create new pull request to `master` now. I think today or tomorrow.

**Comment by fanantoxa** [comment]
@soulcutter I'm closing this task since I should to recreate PR

**Pull request closed without merging**